js/components/search.js

```diff
@@ -5,85 +5,165 @@ $( document ).ready(function() {
+   * @returns {string} glyph and label for content type or generic default
+   */
+  function getContentTypeGlyph(contentType) {
+    var contentTypes = {
+      'text/html':       '<span class="glyphicon glyphicon-link"></span> Website',
+      'application/pdf': '<span class="glyphicon glyphicon-file"></span> PDF',
+      'default':         '<span class="glyphicon glyphicon-file"></span> ' + contentType
+    };
+
+    return contentTypes[contentType]
+      ? contentTypes[contentType]
+      : contentTypes['default']
+  }
+
+  var query = getQueryParam('q'),
+      size  = parseInt(getQueryParam('size'), 10) || 10,
+      from  = parseInt(getQueryParam('from'), 10) || 0;
+
+  $('.search-string').text( query );
+  $('.site-search-text').attr('value', query );
+
+  if ( ! query ) {
     $('.search-no-results').show();
     $('.loading').remove();
     $('.search-results-count').append( 0 + ' search results');
-
+    return;
   }
+
+  $('input.q').val( query );
+  $('.search-result-list').html(
+    '<div class="loading">' +
+      '<span class="glyphicon glyphicon-refresh"></span> Loading' +
+    '</div>'
+  );
+
+  $.ajax({
+    url: apiSearchUrl( query, eiti.beckleyApiKey, {
+      size: size,
+      from: from
+    }),
+    cache: false,
+    dataType: 'json'
+  })
+    .done(function(json) {
+      $('.loading').remove();
+      $('.search-results-count').append( json.hits.total + ' search results');
+
+      if (json.hits.total == 0){
+        $('.search-no-results').show();
+      }
+      else{
+        $('.search-no-results').remove();
+      }
+
+      $.each(json.hits.hits, function(i, hit){
+        var tags = (hit._source.tag || [])
+          .map(function(tag) {
+            return (
+              '<span class="search-result-list-tag">&nbsp;' +
+                '<a href="' + localSearchUrl(tag, {size: size, from: from}) + '" title="Search for ' + tag + '">' +
+                  tag +
+                '</a>&nbsp;' +
+              '</span>'
+            );
+          })
+          .join('');
+
+        var contentType = getContentTypeGlyph(hit._source.content_type);
+
+        $('.search-results-container').append(
+          '<article class="search-result-list">' +
+            '<h1><a href="' + hit._source.url + '" target="_blank">' +
+              hit._source.title +
+            '</a></h1>' +
+            '<p>' + hit._source.description + '</p>' +
+            '<p>Tagged&nbsp;/' + tags + '</p>' +
+          '</article>'
+        );
+      });
+
+      var paging = {
+        needsPrevLink: from > 0,
+        needsNextLink: (from + size) < json.hits.total,
+        urlPrev: localSearchUrl(query, { size: size, from: Math.max(0, from - size - 1) }),
+        urlNext: localSearchUrl(query, { size: size, from: from + size + 1 }),
+        titlePrev: 'Previous page of search results',
+        titleNext: 'Next page of search results'
+      };
+
+      $('.search-results-container .search-results-container').append(
```

Comment:
I'm not sure how this isn't affecting the tag list above, but there are two containers with this same class. To prevent adding links twice, I appended to the innermost of the two.